src/main/java/gyro/google/compute/ProjectMetadataItemFinder.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Metadata;
+import gyro.core.GyroException;
+import gyro.core.Type;
+import gyro.google.GoogleFinder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Query a project-wide metadata item.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    project-metadata-item: $(external-query google::project-metadata-item { key: 'example-key'})
+ */
+@Type("project-metadata-item")
+public class ProjectMetadataItemFinder extends GoogleFinder<Compute, Metadata.Items, ProjectMetadataItemResource> {
+    private String key;
+
+    /**
+     * The key of the metadata item.
+     */
+    public String getKey() {
+        return key;
+    }
+
+    public void setKey(String key) {
+        this.key = key;
+    }
+
+    @Override
+    protected List<Metadata.Items> findAllGoogle(Compute client) {
+        try {
+            return client.projects().get(getProjectId()).execute().getCommonInstanceMetadata().getItems();
+        } catch (GoogleJsonResponseException je) {
+            throw new GyroException(je.getDetails().getMessage());
+        } catch (IOException ex) {
+            throw new GyroException(ex);
+        }
+    }
+
+    @Override
+    protected List<Metadata.Items> findGoogle(Compute client, Map<String, String> filters) {
+        try {
+            Metadata.Items item = client.projects().get(getProjectId()).execute().getCommonInstanceMetadata().getItems().stream()
+                .filter(r -> filters.get("name").equals(r.getKey()))
```

Comment:
`name` should be `key` as that is the field in the class.

running a gyro file containing --> `@print: $(external-query google::project-metadata-item {key: 'ssh-keys'})` results in :
```
Error: Can't process the @print directive!

In compute/project-metadata-item.gyro on line 1 from column 1 to 72:
1: @print: $(external-query google::project-metadata-item {key: 'ssh-keys'})

Caused by: Error: Can't resolve external-query reference!

In compute/project-metadata-item.gyro on line 1 from column 9 to 72:
1: @print: $(external-query google::project-metadata-item {key: 'ssh-keys'})

Caused by: Unexpected error: java.lang.NullPointerException
	at gyro.google.compute.ProjectMetadataItemFinder.lambda$findGoogle$0(ProjectMetadataItemFinder.java:71)
	at gyro.google.compute.ProjectMetadataItemFinder.lambda$findGoogle$0(ProjectMetadataItemFinder.java:71)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1631)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at gyro.google.compute.ProjectMetadataItemFinder.findGoogle(ProjectMetadataItemFinder.java:72)
	at gyro.google.compute.ProjectMetadataItemFinder.findGoogle(ProjectMetadataItemFinder.java:41)
	at gyro.google.GoogleFinder.find(GoogleFinder.java:41)
	at gyro.core.reference.FinderReferenceResolver.resolve(FinderReferenceResolver.java:57)
	at gyro.core.scope.NodeEvaluator.visitReference(NodeEvaluator.java:657)
	at gyro.core.scope.NodeEvaluator.visitReference(NodeEvaluator.java:79)
	at gyro.lang.ast.value.ReferenceNode.accept(ReferenceNode.java:68)
	at gyro.lang.ast.NodeVisitor.visit(NodeVisitor.java:37)
	at gyro.core.directive.DirectiveProcessor.convert(DirectiveProcessor.java:109)
	at gyro.core.directive.DirectiveProcessor.getArgument(DirectiveProcessor.java:117)
	at gyro.core.PrintDirectiveProcessor.process(PrintDirectiveProcessor.java:32)
	at gyro.core.scope.NodeEvaluator.visitDirective(NodeEvaluator.java:344)
	at gyro.core.scope.NodeEvaluator.visitDirective(NodeEvaluator.java:79)
	at gyro.lang.ast.block.DirectiveNode.accept(DirectiveNode.java:85)
	at gyro.lang.ast.NodeVisitor.visit(NodeVisitor.java:37)
	at gyro.core.scope.NodeEvaluator.lambda$evaluateBody$30(NodeEvaluator.java:302)
	at gyro.core.scope.Defer.execute(Defer.java:50)
	at gyro.core.scope.NodeEvaluator.evaluateBody(NodeEvaluator.java:302)
	at gyro.core.scope.NodeEvaluator.visitFile(NodeEvaluator.java:385)
	at gyro.core.scope.NodeEvaluator.visitFile(NodeEvaluator.java:79)
	at gyro.lang.ast.block.FileNode.accept(FileNode.java:32)
	at gyro.lang.ast.NodeVisitor.visit(NodeVisitor.java:37)
	at gyro.core.scope.NodeEvaluator.lambda$evaluateBody$30(NodeEvaluator.java:302)
	at gyro.core.scope.Defer.execute(Defer.java:50)
	at gyro.core.scope.NodeEvaluator.evaluateBody(NodeEvaluator.java:302)
	at gyro.core.scope.NodeEvaluator.evaluate(NodeEvaluator.java:286)
	at gyro.core.scope.RootScope.evaluate(RootScope.java:313)
	at gyro.core.command.AbstractConfigCommand.doExecute(AbstractConfigCommand.java:132)
	at gyro.core.command.AbstractCommand.execute(AbstractCommand.java:71)
	at gyro.cli.Gyro.run(Gyro.java:192)
	at gyro.cli.Gyro.main(Gyro.java:76)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)


```